flowermd/modules/surface_wetting/surface_wetting.py

```diff
@@ -0,0 +1,402 @@
+        wetting_snapshot.dihedrals.typeid = np.concatenate(
+            (
+                self.surface_snapshot.dihedrals.typeid,
+                self.drop_snapshot.dihedrals.typeid
+                + len(self.surface_snapshot.dihedrals.types),
+            ),
+            axis=None,
+        )
+        wetting_snapshot.dihedrals.group = np.concatenate(
+            (
+                self.surface_snapshot.dihedrals.group,
+                self.drop_snapshot.dihedrals.group + self._surface_n,
+            ),
+            axis=None,
+        )
+        # set up pairs
+        wetting_snapshot.pairs.N = (
+            self.surface_snapshot.pairs.N + self.drop_snapshot.pairs.N
+        )
+        # rename surface pair types (add '_' to the beginning)
+        surface_pair_types = []
+        for pair in self.surface_snapshot.pairs.types:
+            p1, p2 = pair.split("-")
+            surface_pair_types.append(f"_{p1}-_{p2}")
+        wetting_snapshot.pairs.types = (
+            surface_pair_types + self.drop_snapshot.pairs.types
+        )
+        wetting_snapshot.pairs.typeid = np.concatenate(
+            (
+                self.surface_snapshot.pairs.typeid,
+                self.drop_snapshot.pairs.typeid
+                + len(self.surface_snapshot.pairs.types),
+            ),
+            axis=None,
+        )
+        wetting_snapshot.pairs.group = np.concatenate(
+            (
+                self.surface_snapshot.pairs.group,
+                self.drop_snapshot.pairs.group + self._surface_n,
+            ),
+            axis=None,
+        )
+
+        return wetting_snapshot
+
+    def _create_box(self):
+        """Create the wetting simulation box."""
+        wetting_sim_box = np.copy(self.surface_snapshot.configuration.box)
+        # use box height for z
+        wetting_sim_box[2] = self.box_height
+        return wetting_sim_box
+
+    def _adjust_particle_positions(self, wetting_box):
+        """Place the surface and droplet particles in the wetting box."""
+        # shift surface particles to the bottom of the box
+        surface_z_shift = (
+            np.abs(
+                min(self.surface_snapshot.particles.position[:, 2])
+                - (-wetting_box[2] / 2)
+            )
+            - 0.1
+        )
+        surface_pos = self.surface_snapshot.particles.position - np.array(
+            [0, 0, surface_z_shift]
+        )
+        # find center of the droplet and shift the droplet particles to origin
+        drop_pos = self.drop_snapshot.particles.position - np.mean(
+            self.drop_snapshot.particles.position, axis=0
+        )
+        # shift drop particles z position to be at the top of surface
+        z_shift = (
+            np.abs(min(drop_pos[:, 2]) - max(surface_pos[:, 2])) - self.gap
+        )
+        drop_pos[:, 2] -= z_shift
+        wetting_pos = np.concatenate((surface_pos, drop_pos), axis=0)
+        return wetting_pos
+
+    @property
+    def wetting_snapshot(self):
```

Comment:
For consistency purposes, I think we should call this property `hoomd_snapshot`. That is what we use in `System` and what I was using as an attribute in the welding module.